src/libs/ajax/AzureStorage.ts

```diff
@@ -131,10 +131,15 @@ export const AzureStorage = (signal?: AbortSignal) => ({
         const urlwithFolder = new URL(azureStorageUrl);
         const azureSasStorageUrl = `https://${urlwithFolder.hostname}${urlwithFolder.pathname}?${token}`;
 
-        const res = await fetchOk(azureSasStorageUrl, { method: 'GET' });
+        const res = await fetchOk(azureSasStorageUrl, { method });
         const headerDict = Object.fromEntries(res.headers);
 
-        const textContent = await res.text().catch((_err) => undefined);
+        let textContent;
+        if (method === 'GET') {
+          textContent = await res.text().catch((_err) => undefined);
+        } else {
+          textContent = null;
+        }
```

Comment:
"at least then it would be constant in the rest of the function" is something I hadn't considered... I wrote it as you suggested originally, but I've been nudged away from ternary operators in the past (though maybe I overgeneralized that lesson, I'm not sure)